### ISSUE TITLE
When scanning apple/swift, linting one file fails with a segfault

### ISSUE THREAD

**Issue by mildm8nnered** [realm/SwiftLint#4830]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

```
brew install swiftlint
git clone https://github.com/apple/swift
cd swift
swiftlint ./test/Parse/operator_decl.swift
```

```
Linting Swift files at paths ./test/Parse/operator_decl.swift
Linting 'operator_decl.swift' (1/1)
PLEASE submit a bug report to https://bugs.llvm.org/ and include the crash backtrace.
zsh: segmentation fault  swiftlint ./test/Parse/operator_decl.swift.```
```

Only `./test/Parse/operator_decl.swift` is affected.

This also looks like it may be a SourceKit related issue:

<img width="1809" alt="Screenshot 2023-03-19 at 13 03 24" src="https://user-images.githubusercontent.com/435558/226176974-eb8a6d18-76f9-43b3-8858-11b510bf4507.png">

##### Complete output when running SwiftLint, including the stack trace and command used

See above

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? `0.50.3`
* Installation method used (Homebrew, CocoaPods, building from source, etc)? homebrew
* Paste your configuration file:

none

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? no
* Which Xcode version are you using (check `xcodebuild -version`)? 

```
Xcode 14.2
Build version 14C18
```

* Do you have a sample that shows the issue? 

See above for sample files that cause the issue

**Comment by mildm8nnered** [comment]
[operator_decl.swift.zip](https://github.com/realm/SwiftLint/files/11011084/operator_decl.swift.zip)

**Comment by mildm8nnered** [comment]
This is fixed, at least as of 0.57.0. Yay!
